esx_service/vmdk_ops.py

```diff
@@ -314,15 +314,25 @@ def find_child(vm_name):
 
 #returns error, or None for OK
 def attachVMDK(vmdk_path, vm_name):
-    vm = findVmByName(vm_name)
+    try:
+       vm = findVmByName(vm_name)
+    except Exception as e:
+       msg = "VM {0} not found".format(vm_name)
+       return err(msg)
+
     logging.info("*** attachVMDK: %s to %s VM uuid = %s",
                  vmdk_path, vm_name, vm.config.uuid)
     return disk_attach(vmdk_path, vm)
 
 
 #returns error, or None for OK
 def detachVMDK(vmdk_path, vm_name):
-    vm = findVmByName(vm_name)
+    try:
+       vm = findVmByName(vm_name)
+    except Exception as e:
```

Comment:
I'd prefer that we don't raise exceptions anywhere in the plugin code and
instead handle the exception at the point that we could get one and
translate to an appropriate error return. In which case findVmByName()
mustn't be raising exceptions.

On Thu, Jun 2, 2016 at 12:49 PM, Mark Sterin <notifications@github.com>
wrote:

> In esx_service/vmdk_ops.py
> <https://github.com/vmware/docker-volume-vsphere/pull/433#discussion_r65494072>
> :
>
> >      logging.info("*** attachVMDK: %s to %s VM uuid = %s",
> >                   vmdk_path, vm_name, vm.config.uuid)
> >      return disk_attach(vmdk_path, vm)
> >
> >
> >  #returns error, or None for OK
> >  def detachVMDK(vmdk_path, vm_name):
> > -    vm = findVmByName(vm_name)
> > +    try:
> > +       vm = findVmByName(vm_name)
> > +    except Exception as e:
>
> I agree with Prashant - it looks strange to mix-n-match exceptions and
> return code without any visible system
> We can agree on "catch exceptoins as soon as they happen and pass error
> code around" or "catch exceptions only when decision needs to be made".
> findVmByName handling mixes both - I think it needs to be changed to a
> single style
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/433/files/1ae935fbbc0c3205389f8fd83888b9afea83cba2#r65494072>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHseAQEce9ewGFizUGGTpVPtN5KqMSdks5qHoPugaJpZM4IrROI>
> .
>